vendor/github.com/openshift/library-go/pkg/operator/status/controller.go

```diff
@@ -0,0 +1,199 @@
+package status
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/golang/glog"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+
+	operatorv1alpha1 "github.com/openshift/api/operator/v1alpha1"
+	"github.com/openshift/library-go/pkg/operator/v1alpha1helpers"
+)
+
+var workQueueKey = "instance"
+
+type OperatorStatusProvider interface {
+	Informer() cache.SharedIndexInformer
+	CurrentStatus() (operatorv1alpha1.OperatorStatus, error)
+}
+
+type StatusSyncer struct {
+	clusterOperatorNamespace string
+	clusterOperatorName      string
+
+	// TODO use a generated client when it moves to openshift/api
+	clusterOperatorClient dynamic.ResourceInterface
+
+	operatorStatusProvider OperatorStatusProvider
+
+	// queue only ever has one item, but it has nice error handling backoff/retry semantics
+	queue workqueue.RateLimitingInterface
+}
+
+func NewClusterOperatorStatusController(
+	namespace, name string,
+	clusterOperatorClient dynamic.Interface,
+	operatorStatusProvider OperatorStatusProvider,
+) *StatusSyncer {
+	c := &StatusSyncer{
+		clusterOperatorNamespace: namespace,
+		clusterOperatorName:      name,
+		clusterOperatorClient:    clusterOperatorClient.Resource(schema.GroupVersionResource{Group: "operatorstatus.openshift.io", Version: "v1", Resource: "ClusterOperator"}).Namespace(namespace),
+		operatorStatusProvider:   operatorStatusProvider,
+
+		queue: workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "StatusSyncer-"+name),
+	}
+
+	operatorStatusProvider.Informer().AddEventHandler(c.eventHandler())
+	// TODO watch clusterOperator.status changes when it moves to openshift/api
+
+	return c
+}
+
+// sync reacts to a change in prereqs by finding information that is required to match another value in the cluster. This
+// must be information that is logically "owned" by another component.
+func (c StatusSyncer) sync() error {
+	currentDetailedStatus, err := c.operatorStatusProvider.CurrentStatus()
+	if apierrors.IsNotFound(err) {
+		return c.clusterOperatorClient.Delete(c.clusterOperatorName, nil)
+	}
+	if err != nil {
+		return err
+	}
+
+	operatorConfig, err := c.clusterOperatorClient.Get(c.clusterOperatorName, metav1.GetOptions{})
```

Comment:
odd the get the same object again.

Reply:
> odd the get the same object again.

Uncommon, but dynamic informers/listers are in 1.13 and this is fairly slow moving.